libtenzir/builtins/aggregation-functions/collect.cpp

```diff
@@ -0,0 +1,79 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2024 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <tenzir/aggregation_function.hpp>
+#include <tenzir/plugin.hpp>
+
+namespace tenzir::plugins::collect {
+
+namespace {
+
+template <concrete_type Type>
+class collect_function final : public aggregation_function {
+public:
+  explicit collect_function(type input_type) noexcept
+    : aggregation_function(std::move(input_type)) {
+    // nop
+  }
+
+private:
+  auto output_type() const -> type override {
+    return type{list_type{input_type()}};
+  }
+
+  auto add(const data_view& view) -> void override {
+    using view_type = tenzir::view<type_to_data_t<Type>>;
+    if (caf::holds_alternative<caf::none_t>(view)) {
+      return;
+    }
```

Comment:
We may want to make treating nulls a runtime option, but we can also wait until the first user requests that.

Reply:
I don't think it'd be particularly useful tbh, so let's definitely wait until someone asks for it.